infra/charts/feast-feature-server/templates/deployment.yaml

```diff
@@ -33,19 +33,46 @@ spec:
           env:
             - name: FEATURE_STORE_YAML_BASE64
               value: {{ .Values.feature_store_yaml_base64 }}
-          command: ["feast", "serve", "-h", "0.0.0.0"]
+          command:
+            {{- if eq .Values.feast_mode "offline" }}
+            - "feast"
+            - "serve_offline"
+            - "-h"
+            - "0.0.0.0"
+            {{- else if eq .Values.feast_mode "ui" }}
+            - "feast"
+            - "ui"
+            - "-h"
+            - "0.0.0.0"
+            {{- else if eq .Values.feast_mode "registry" }}
+            - "feast"
+            - "serve_registry"
+            {{- else }}
+            - "feast"
+            - "serve"
+            - "-h"
+            - "0.0.0.0"
+            {{- end }}
           ports:
-            - name: http
+            - name: {{ .Values.feast_mode }}
```

Comment:
Do we really need a different `containerPort` for server? Different pods and containers can have the same `containerPort` IIRC.

Why not adding a value `containerPort: 8888` (any value) and include `-p {{ .Values.containerPort }}` in each command section to configure it?
ATM this configuration works only because the following `containerPort`s match the defaults that are used in `cli.py`, but if they change the chart would fail to deploy the services.

Reply:
my 2 cents, i kind of like the way its currently coded. the service's `targetPort` will also change based on `feast_mode` and this gives the user an easy way to identify that the helm chart has succeeded, and that feast is running the mode they intended.

Reply:
using the default ports also removes any guesswork from the user who is trying to connect via service port forwarding or debugging

Reply:
🤔 IIUC, service.targetPort reflects the feast_mode, not the container port, and when you port forward the service, you always run the same command whatever the server type, because the service port is set to `80`:
`oc port-forward svc/feast-release-feast-feature-server LOCAL_PORT:80`

If we don't want to use the same port, I'd suggest at least to use the `-p` option to set the server's port accordingly:
```yaml
            {{- if eq .Values.feast_mode "offline" }}
            - "feast"
            - "serve_offline"
            - "-h"
            - "0.0.0.0"
            - "-p"
            - "8815"
           ...
```

Reply:
its my understanding that the `targetPort` reflects the name or int of the container port. that said, you're correct about the port forwarding because it will remain port 80 regardless. regardless, IMO... i like the idea of using the default ports for each mode type and reflecting that change in the service `targetPort`. just my opinion.

Reply:
its worth noting that the user can change from the default port 80 by setting the `service.port` param during helm install

Reply:
@tchughesiv IIUC, you propose to:
- apply `-p` option for each mode type
- replace `targetPort` definition with the same value as the containerPort for the selected server
(so when we list `oc get svc` it will show the actual port number instead of just `80/TCP`)

Reply:
I have updated the command by passing `-p` option for the port.

Reply:
i think you can leave off the `-p` ... lets just use the default ports for each type. no, @dmartinol i'm good with the PR as-is. the `targetPort` will already reflect the `feast_mode` as its coded.

Reply:
okay keeping same and removed -`p` from the command

Reply:
@redhatHameed could you pls deploy the 4 services and share the output of `oc get svc`?